src/engine/TPA.cc

```diff
@@ -73,8 +74,17 @@ VerificationResult TPAEngine::solve(const ChcDirectedGraph & graph) {
                 assert(false);
                 throw std::logic_error("Unreachable!");
         }
-    } else if (isTransitionSystemDAG(graph) && not options.hasOption(Options::FORCE_TS)) {
-        return solveTransitionSystemGraph(graph);
+    } else if ((isTransitionSystemDAG(graph) && not options.hasOption(Options::FORCE_TS)) ||
+               options.hasOption(Options::SIMPLIFY_NESTED)) {
+        if (options.hasOption(Options::SIMPLIFY_NESTED)) {
+            NestedLoopTransformation transformation;
+            auto normalGraph = graph;
```

Comment:
You don't need a copy here, no?

Reply:
Nope, I need it bc `graph` is a const, and I'm changing `normalGraph` inside of the `transform` function

Reply:
I also need a const version inside of `transform` to store in the witness

Reply:
You can change `transform` to take `const &`, no?
Since you are making a copy inside anyway, you can then operate on the copy.

Reply:
Ok, didn't want to return modified copy, will do

Reply:
Fixed